arch/arm64/boot/dts/qcom/msm8953-huawei-milan.dts

```diff
@@ -0,0 +1,618 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (c) 2023, Clrs17<2318678921@qq.com>
+ * Copyright (c) 2023, bzy-080408<Bzy080408@outlook.com>
+ */
+/dts-v1/;
+
+#include "msm8953.dtsi"
+#include "pm8953.dtsi"
+#include "pmi8950.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/interrupt-controller/irq.h>
+#include <dt-bindings/leds/common.h>
+
+
+/ {
+	model = "Huawei Maimang 5 / Nova (Plus) / G9 (Plus)";
+	compatible = "huawei,milan", "qcom,msm8953", "lk2nd,device";
+	qcom,msm-id = <293 0x0>;
+	qcom,board-id = <8000 0>, <8001 0>, <8002 0>, <8003 0>, <8016 0>,
+			<8017 0>, <8018 0>, <8019 0>, <8032 0>, <8033 0>,
+			<8035 0>, <8048 0>, <8049 0>, <8050 0>, <8051 0>,
+			<8066 0>, <8064 0>, <8067 0>, <8080 0>, <8082 0>,
+			<8083 0>, <8098 0>, <8099 0>, <8114 0>, <8130 0>;
+	chassis-type = "handset";
+	qcom,msm-id = <293 0>;
+	
+	battery: battery {
+		compatible = "simple-battery";
+		charge-full-design-microamp-hours = <0x31e570>;
+		voltage-min-design-microvolt = <0x33e140>;
+		voltage-max-design-microvolt = <0x42d560>;
+		phandle = <0x53>;
+	};
+	
+	vph_pwr: vph-pwr-regulator {
+		compatible = "regulator-fixed";
+		regulator-name = "vph_pwr";
+		regulator-always-on;
+		regulator-boot-on;
+	};
+
+	chosen {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		framebuffer@90001000 {
+			compatible = "simple-framebuffer";
+			reg = <0x00 0x90001000 0x00 0x5eec00>;
+			width = <0x438>;
+			height = <0x780>;
+			stride = <0xca8>;
+			format = "r8g8b8";
+
+			power-domains = <&gcc MDSS_GDSC>;
+
+			clocks = <&gcc GCC_MDSS_AHB_CLK>,
+					 <&gcc GCC_MDSS_AXI_CLK>,
+					 <&gcc GCC_MDSS_VSYNC_CLK>,
+					 <&gcc GCC_MDSS_MDP_CLK>,
+					 <&gcc GCC_MDSS_BYTE0_CLK>,
+					 <&gcc GCC_MDSS_PCLK0_CLK>,
+					 <&gcc GCC_MDSS_ESC0_CLK>;
+		};
+	};
+
+	gpio-keys {
+		compatible = "gpio-keys";
+		pinctrl-names = "default";
+		pinctrl-0 = <0xbb>;
+
+		key-volume-up {
+			label = "volume_up";
+			linux,code = <0x73>;
+			gpios = <0x2d 0x55 0x01>;
+		};
+	};
+
+
+
+	reserved-memory {
```

Comment:
You only need to keep removed-dma-pools what is differs from the reference in downstream, everything else could be removed.